src/components/FacetProps.tsx

```diff
@@ -83,9 +91,33 @@ export interface NumericalFacetProps extends StandardFacetProps {
   inputPrefix?: JSX.Element
 }
 
+/**
+ * Props for the {@link StandardFacet} component.
+ *
+ * @public
+ */
+export interface HierarchicalFacetProps extends
+  Omit<StandardFacetProps, 'transformOptions' | 'showOptionCounts'> {
+  /**
+   * A function to transform facet's options. The returned options need to be delimited to keep
+   * the hierarchy.
+   */
+  transformOptions?: (options: DisplayableFacetOption[]) => DisplayableFacetOption[],
+  /**
+   * The maximum number of options to render before displaying the "Show more/less" button.
+   * Defaults to 4.
+   */
+  showMoreLimit?: number,
+  /** CSS classes for customizing the component styling. */
+  customCssClasses?: Omit<FilterGroupCssClasses, 'searchInput' | 'optionsContainer' | 'option' |
+  'optionInput' | 'optionLabel'> & HierarchicalFacetDisplayCssClasses,
```

Comment:
I recommend that we instead have a single `HierarchicalFacetDisplayCssClasses` interface because using `Omit` and `&` doesn't lead to clear auto-generated documentation. That would lead to some duplicated properties, but we can use `@inheritDoc` to avoid duplicating the property descriptions.

Reply:
I was trying to make it clear that it derives from FilterGroupCssClasses. But, yeah, as there are less props in common, it's better to make a separate props.